.github/workflows/run-tests.yml

```diff
@@ -292,6 +292,7 @@ jobs:
           echo "TH_SNOW_URL=jdbc:snowflake://http://snowflake.localhost.localstack.cloud:4566" >> $GITHUB_ENV
 
       - name: Run Tests
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' && matrix.testSystem == 'snowflake' }}
```

Comment:
I don't think this is what you want either, because this is going to run the tests ONLY on master and ONLY on snowflake.

You want something like:

`if: master || !snowflake`

Unrelated, but, will the push event trigger when a PR is merged to master? I am unfamiliar.

Reply:
hmm, it should be what I had previously ie `pull_request`

Reply:
I don't think you need any condition related to the `event_name`. The condition I put in my previous comment should be sufficient. Run only on master or, if not on master, run only if not in the snowflake part of the matrix.

Reply:
we do need to take into consideration if its' a PR or a merge to master though, hence the `event_name`

Reply:
Wouldn't `github.ref` take care of that?

Reply:
what you mentioned translates to `if: github.ref == 'refs/heads/master' || matrix.testSystem != 'snowflake'`

Reply:
I think we should test it out